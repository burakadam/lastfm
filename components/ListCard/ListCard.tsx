import Image from 'next/image';
import { listeners } from 'process';

interface IListCardProps {
  name: string;
  title: string;
  image: string;
  playcount: string;
  listeners?: string;
}

const ListCard = ({
  name,
  title,
  image,
  playcount,
  listeners,
}: IListCardProps) => (
  <div className='border p-2 flex mb-2'>
    <div className='mr-2'>
      <Image
        src={image}
        alt={name}
        width={500}
        height={500}
        className='w-12 h-12'
      />
    </div>
    <div className='flex justify-between text-sm flex-1'>
      <div className='mr-2'>
        <h3>{name}</h3>
        <h4 className='font-bold'>{title}</h4>
      </div>
      <div className='flex flex-col justify-center text-xs'>
        {listeners && <p>{listeners} listeners</p>}
        <p>{playcount} play</p>
      </div>
    </div>
  </div>
);

export default ListCard;
