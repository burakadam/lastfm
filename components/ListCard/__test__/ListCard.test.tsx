import { render } from '@testing-library/react';

import ListCard from '../ListCard';

const MOCK_CARD = {
  artist: 'Taylor Swift',
  name: 'Love Story',
  playcount: '8215136',
  listeners: '947798',
  image:
    'https://lastfm.freetls.fastly.net/i/u/64s/2a96cbd8b46e442fc41c2b86b821562f.png',
  index: '1',
};

describe('<ListCard> spec', () => {
  let CARD: any;

  beforeEach(() => {
    CARD = render(
      <ListCard
        key={MOCK_CARD.index}
        name={MOCK_CARD.artist}
        title={MOCK_CARD.name}
        playcount={MOCK_CARD.playcount}
        image={MOCK_CARD.image}
        listeners={MOCK_CARD.listeners}
      />
    );
  });

  it('render a artist list component', () => expect(CARD).toMatchSnapshot());

  it('render artist name', () =>
    expect(CARD.getByText(MOCK_CARD.artist)).toBeTruthy());

  it('render album or track name', () =>
    expect(CARD.getByText(MOCK_CARD.name)).toBeTruthy());

  it('render album or track play', () =>
    expect(CARD.getByText(`${MOCK_CARD.playcount} play`)).toBeTruthy());

  it('render track listeners', () =>
    expect(CARD.getByText(`${MOCK_CARD.listeners} listeners`)).toBeTruthy());
});
