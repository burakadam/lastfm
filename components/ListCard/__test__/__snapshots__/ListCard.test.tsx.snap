// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListCard> spec render a artist list component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border p-2 flex mb-2"
      >
        <div
          class="mr-2"
        >
          <img
            alt="Taylor Swift"
            class="w-12 h-12"
            data-nimg="1"
            decoding="async"
            height="500"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=1080&q=75"
            srcset="/_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=1080&q=75 2x"
            style="color: transparent;"
            width="500"
          />
        </div>
        <div
          class="flex justify-between text-sm flex-1"
        >
          <div
            class="mr-2"
          >
            <h3>
              Taylor Swift
            </h3>
            <h4
              class="font-bold"
            >
              Love Story
            </h4>
          </div>
          <div
            class="flex flex-col justify-center text-xs"
          >
            <p>
              947798
               listeners
            </p>
            <p>
              8215136
               play
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border p-2 flex mb-2"
    >
      <div
        class="mr-2"
      >
        <img
          alt="Taylor Swift"
          class="w-12 h-12"
          data-nimg="1"
          decoding="async"
          height="500"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=1080&q=75"
          srcset="/_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=1080&q=75 2x"
          style="color: transparent;"
          width="500"
        />
      </div>
      <div
        class="flex justify-between text-sm flex-1"
      >
        <div
          class="mr-2"
        >
          <h3>
            Taylor Swift
          </h3>
          <h4
            class="font-bold"
          >
            Love Story
          </h4>
        </div>
        <div
          class="flex flex-col justify-center text-xs"
        >
          <p>
            947798
             listeners
          </p>
          <p>
            8215136
             play
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
